File: backend/app/api/v1/endpoints/insights.py
# =============================================================================
# Stratum AI - Insights & Recommendations API Router
# =============================================================================
"""
API endpoints for Intelligence Layer features:
- Insights (aggregated daily view)
- Recommendations (actionable suggestions)
- Anomaly alerts
"""

from datetime import date, datetime, timedelta, timezone
from typing import Any, Dict, List, Optional

from fastapi import APIRouter, Depends, HTTPException, Query, Request
from sqlalchemy import func, select
from sqlalchemy.ext.asyncio import AsyncSession

from app.analytics.logic.recommend import (
    RecommendationsEngine,
    generate_recommendations,
)
from app.analytics.logic.types import BaselineMetrics, EntityMetrics
from app.auth.deps import get_current_user
from app.db.session import get_async_session
from app.features.service import can_access_feature, get_org_features
from app.quality.trust_layer_service import SignalHealthService
from app.schemas.response import APIResponse

# NOTE(STRAT-SC-001/C6): router-level auth — the old TenantMiddleware 401'd
# every non-public request; AuthContextMiddleware (C2) only decodes the JWT,
# so these analytics reads were reachable anonymously. Same fail-open class
# C3 closed on 11 sibling routers.
router = APIRouter(tags=["insights"], dependencies=[Depends(get_current_user)])


# =============================================================================
# Helper Functions
# =============================================================================


async def get_entity_metrics(
    db: AsyncSession, target_date: date
) -> List[EntityMetrics]:
    """
    Fetch entity metrics for recommendations.
    Queries campaign data for the target date.
    """
    from app.models import Campaign

    # The recommendations engine needs per-campaign rows, but this must not scan
    # an unbounded number of them [API-002]. Cap at the 1000 highest-spend
    # campaigns — the ones that actually drive recommendations — so an
    # organization with a very large campaign count can't load its whole table
    # into memory. No effect for the vast majority of deployments (< 1000
    # campaigns).
    result = await db.execute(
        select(Campaign)
        .where(
            Campaign.is_deleted == False,
        )
        .order_by(Campaign.total_spend_cents.desc().nullslast())
        .limit(1000)
    )
    campaigns = result.scalars().all()

    metrics = []
    for c in campaigns:
        spend = (c.total_spend_cents or 0) / 100
        revenue = (c.revenue_cents or 0) / 100
        metrics.append(
            EntityMetrics(
                entity_id=str(c.id),
                entity_name=c.name or f"Campaign {c.id}",
                entity_type="campaign",
                platform=c.platform.value if c.platform else "unknown",
                spend=spend,
                revenue=revenue,
                roas=revenue / spend if spend > 0 else 0,
                cpa=(
                    (spend / c.conversions)
                    if c.conversions and c.conversions > 0
                    else 0
                ),
                impressions=c.impressions or 0,
                clicks=c.clicks or 0,
                conversions=c.conversions or 0,
                ctr=c.ctr or 0,
            )
        )

    return metrics


async def get_baseline_metrics(
    db: AsyncSession,
) -> Dict[str, BaselineMetrics]:
    """
    Fetch baseline metrics for entities.
    Calculates baselines from historical campaign performance.
    """
    from app.models import Campaign

    active_campaigns = Campaign.is_deleted == False

    # Portfolio-level baseline is a single set of aggregates over all
    # campaigns — compute it in SQL rather than loading every row to sum in
    # Python [API-002].
    agg = (
        await db.execute(
            select(
                func.count(Campaign.id),
                func.coalesce(func.sum(Campaign.total_spend_cents), 0),
                func.coalesce(func.sum(Campaign.revenue_cents), 0),
                func.coalesce(func.sum(Campaign.conversions), 0),
                func.coalesce(func.sum(Campaign.impressions), 0),
                func.coalesce(func.sum(Campaign.clicks), 0),
            ).where(active_campaigns)
        )
    ).one()

    (
        n,
        spend_cents,
        revenue_cents,
        total_conversions,
        total_impressions,
        total_clicks,
    ) = agg
    if n == 0:
        return {}

    total_spend = spend_cents / 100
    total_revenue = revenue_cents / 100

    avg_spend = total_spend / n if n > 0 else 0
    avg_roas = total_revenue / total_spend if total_spend > 0 else 0
    avg_cpa = total_spend / total_conversions if total_conversions > 0 else 0
    avg_ctr = total_clicks / total_impressions * 100 if total_impressions > 0 else 0

    # Every campaign shares the same portfolio baseline; fetch just the ids
    # (not full rows) to key the map.
    ids = (
        (await db.execute(select(Campaign.id).where(active_campaigns))).scalars().all()
    )

    return {
        str(campaign_id): BaselineMetrics(
            avg_spend=avg_spend,
            avg_roas=avg_roas,
            avg_cpa=avg_cpa,
            avg_ctr=avg_ctr,
        )
        for campaign_id in ids
    }


async def check_signal_health_for_autopilot(
    db: AsyncSession, target_date: date
) -> Dict[str, Any]:
    """Check if autopilot should be blocked due to signal health."""
    service = SignalHealthService(db)
    health_data = await service.get_signal_health(target_date)

    status = health_data.get("status", "unknown")
    blocked = status in ["degraded", "critical"]
    reason = None

    if blocked:
        reason = f"Signal health is {status}. Automation blocked for data quality protection."

    return {
        "blocked": blocked,
        "reason": reason,
        "status": status,
    }


async def detect_campaign_anomalies(
    db: AsyncSession,
    target_date: date,
) -> List[Dict[str, Any]]:
    """
    Detect anomalies across campaigns. Pure function — no auth /
    feature-gate / response wrapping. Reused by the `/anomalies`
    endpoint and the `/console/anomalies-rollup` aggregator so the
    detection logic stays in one place.
    """
    from app.models import Campaign

    # Bounded scan [API-002]: check the 1000 highest-spend campaigns rather than
    # the entire table. Anomalies on top-spend campaigns are the ones that
    # matter; no effect for deployments with < 1000 campaigns.
    result = await db.execute(
        select(Campaign)
        .where(
            Campaign.is_deleted == False,
        )
        .order_by(Campaign.total_spend_cents.desc().nullslast())
        .limit(1000)
    )
    campaigns = result.scalars().all()

    anomalies: List[Dict[str, Any]] = []
    anomaly_idx = 0
    detected_at = datetime.now(timezone.utc).isoformat()

    for c in campaigns:
        spend = (c.total_spend_cents or 0) / 100
        revenue = (c.revenue_cents or 0) / 100
        roas = revenue / spend if spend > 0 else 0
        cpa = spend / c.conversions if c.conversions and c.conversions > 0 else 0

        if spend > 0 and roas < 1.0:
            anomaly_idx += 1
            anomalies.append(
                {
                    "id": f"anomaly_{target_date.isoformat()}_{anomaly_idx}",
                    "detected_at": detected_at,
                    "metric": "roas",
                    "entity_type": "campaign",
                    "entity_id": str(c.id),
                    "entity_name": c.name or f"Campaign {c.id}",
                    "severity": "critical" if roas < 0.5 else "high",
                    "direction": "drop",
                    "current_value": round(roas, 2),
                    "expected_value": None,
                    "description": f"ROAS at {roas:.2f}x is below break-even threshold",
                    "possible_causes": [
                        "Audience fatigue",
                        "Increased competition",
                        "Poor creative performance",
                    ],
                    "recommended_actions": [
                        "Review targeting",
                        "Refresh creatives",
                        "Consider pausing",
                    ],
                }
            )

        if c.conversions and c.conversions > 0 and cpa > 100:
            anomaly_idx += 1
            anomalies.append(
                {
                    "id": f"anomaly_{target_date.isoformat()}_{anomaly_idx}",
                    "detected_at": detected_at,
                    "metric": "cpa",
                    "entity_type": "campaign",
                    "entity_id": str(c.id),
                    "entity_name": c.name or f"Campaign {c.id}",
                    "severity": "medium",
                    "direction": "spike",
                    "current_value": round(cpa, 2),
                    "expected_value": 50.0,
                    "description": f"CPA at ${cpa:.2f} is significantly above target",
                    "possible_causes": [
                        "Low conversion rate",
                        "High CPC",
                        "Landing page issues",
                    ],
                    "recommended_actions": [
                        "Optimize landing page",
                        "Narrow targeting",
                        "Test new creatives",
                    ],
                }
            )

    return anomalies


# =============================================================================
# Insights Endpoint
# =============================================================================


@router.get("/insights", response_model=APIResponse[Dict[str, Any]])
async def get_insights(
    request: Request,
    target_date: Optional[date] = Query(default=None, alias="date"),
    db: AsyncSession = Depends(get_async_session),
):
    """
    Get daily insights.

    Returns aggregated view of:
    - KPIs and trends
    - Top actions/recommendations
    - Risks and opportunities
    - Autopilot status

    Requires feature flag: ai_recommendations
    """
    if not await can_access_feature(db, "ai_recommendations"):
        raise HTTPException(
            status_code=403,
            detail="AI recommendations feature is not enabled",
        )

    if target_date is None:
        target_date = date.today()

    # Get org features for autopilot level
    features = await get_org_features(db)
    autopilot_level = features.get("autopilot_level", 0)

    # Check signal health for autopilot blocking
    autopilot_status = await check_signal_health_for_autopilot(db, target_date)

    # Get metrics (placeholder data for now)
    entities_today = await get_entity_metrics(db, target_date)
    baselines = await get_baseline_metrics(db)

    # Generate recommendations if we have data
    if entities_today and baselines:
        engine = RecommendationsEngine()
        recommendations_data = engine.generate_recommendations(
            entities_today=entities_today,
            baselines=baselines,
        )
    else:
        # Return placeholder insights when no data
        recommendations_data = {
            "recommendations": [],
            "actions": [],
            "alerts": [],
            "insights": [],
            "health": {"status": "no_data"},
            "scaling_summary": {
                "scale_candidates": 0,
                "fix_candidates": 0,
                "watch_candidates": 0,
            },
            "generated_at": None,
            "automation_blocked": autopilot_status["blocked"],
        }

    # Build insights response
    response = {
        "date": target_date.isoformat(),
        "kpis": {
            "total_spend": 0,
            "total_revenue": 0,
            "roas": 0,
            "cpa": 0,
            "trend_vs_yesterday": 0,
            "trend_vs_last_week": 0,
        },
        "actions": recommendations_data.get("recommendations", [])[:5],  # Top 5 actions
        "risks": [
            alert
            for alert in recommendations_data.get("alerts", [])
            if alert.get("severity") in ["high", "critical"]
        ],
        "opportunities": recommendations_data.get("insights", [])[
            :3
        ],  # Top 3 opportunities
        "autopilot": {
            "level": autopilot_level,
            "level_name": {
                0: "Suggest Only",
                1: "Guarded Auto",
                2: "Approval Required",
            }.get(autopilot_level, "Unknown"),
            "blocked": autopilot_status["blocked"],
            "reason": autopilot_status["reason"],
        },
        "signal_health_status": autopilot_status["status"],
        "scaling_summary": recommendations_data.get("scaling_summary", {}),
    }

    return APIResponse(success=True, data=response)


# =============================================================================
# Recommendations Endpoint
# =============================================================================


@router.get("/recommendations", response_model=APIResponse[Dict[str, Any]])
async def get_recommendations(
    request: Request,
    target_date: Optional[date] = Query(default=None, alias="date"),
    entity_type: Optional[str] = Query(
        default=None, description="Filter by entity type: campaign, adset, creative"
    ),
    priority: Optional[str] = Query(
        default=None, description="Filter by priority: critical, high, medium, low"
    ),
    limit: int = Query(default=20, ge=1, le=100),
    db: AsyncSession = Depends(get_async_session),
):
    """
    Get detailed recommendations.

    Each recommendation includes:
    - type: budget_shift, creative_refresh, fix_campaign, etc.
    - entity: campaign/adset/creative with ID and name
    - title: human-readable action title
    - why: list of reasons/factors
    - confidence: 0-1 confidence score
    - risk: low/medium/high
    - guardrails: any caps or limits applied
    """
    if not await can_access_feature(db, "ai_recommendations"):
        raise HTTPException(
            status_code=403,
            detail="AI recommendations feature is not enabled",
        )

    if target_date is None:
        target_date = date.today()

    # Get metrics
    entities_today = await get_entity_metrics(db, target_date)
    baselines = await get_baseline_metrics(db)

    # Generate recommendations
    if entities_today and baselines:
        engine = RecommendationsEngine()
        recommendations_data = engine.generate_recommendations(
            entities_today=entities_today,
            baselines=baselines,
        )
        recommendations = recommendations_data.get("recommendations", [])
    else:
        recommendations = []

    # Apply filters
    if entity_type:
        recommendations = [
            r for r in recommendations if r.get("entity_type") == entity_type
        ]

    if priority:
        recommendations = [r for r in recommendations if r.get("priority") == priority]

    # Apply limit
    recommendations = recommendations[:limit]

    # Add guardrails info to each recommendation
    from app.features.flags import get_autopilot_caps

    caps = get_autopilot_caps()

    for rec in recommendations:
        rec["guardrails"] = {
            "max_budget_change_pct": caps["max_budget_pct_change"],
            "max_daily_budget_change": caps["max_daily_budget_change"],
            "requires_approval": caps.get("requires_approval", False),
        }

    return APIResponse(
        success=True,
        data={
            "date": target_date.isoformat(),
            "recommendations": recommendations,
            "total": len(recommendations),
            "filters_applied": {
                "entity_type": entity_type,
                "priority": priority,
            },
        },
    )


# =============================================================================
# Anomalies Endpoint
# =============================================================================


@router.get("/anomalies", response_model=APIResponse[Dict[str, Any]])
async def get_anomalies(
    request: Request,
    target_date: Optional[date] = Query(default=None, alias="date"),
    days: int = Query(
        default=7, ge=1, le=30, description="Days to look back for anomaly detection"
    ),
    severity: Optional[str] = Query(
        default=None, description="Filter by severity: critical, high, medium, low"
    ),
    db: AsyncSession = Depends(get_async_session),
):
    """
    Get anomaly alerts.

    Detects unusual patterns in:
    - Spend spikes/drops
    - ROAS changes
    - CPA anomalies
    - Conversion rate shifts

    Requires feature flag: anomaly_alerts
    """
    if not await can_access_feature(db, "anomaly_alerts"):
        raise HTTPException(
            status_code=403,
            detail="Anomaly alerts feature is not enabled",
        )

    if target_date is None:
        target_date = date.today()

    anomalies = await detect_campaign_anomalies(db, target_date)

    if severity:
        anomalies = [a for a in anomalies if a.get("severity") == severity]

    return APIResponse(
        success=True,
        data={
            "date": target_date.isoformat(),
            "lookback_days": days,
            "anomalies": anomalies,
            "total": len(anomalies),
            "by_severity": {
                "critical": len(
                    [a for a in anomalies if a.get("severity") == "critical"]
                ),
                "high": len([a for a in anomalies if a.get("severity") == "high"]),
                "medium": len([a for a in anomalies if a.get("severity") == "medium"]),
                "low": len([a for a in anomalies if a.get("severity") == "low"]),
            },
        },
    )


# =============================================================================
# KPIs Endpoint
# =============================================================================


@router.get("/kpis", response_model=APIResponse[Dict[str, Any]])
async def get_kpis(
    request: Request,
    target_date: Optional[date] = Query(default=None, alias="date"),
    comparison: str = Query(
        default="yesterday",
        description="Comparison period: yesterday, last_week, last_month",
    ),
    db: AsyncSession = Depends(get_async_session),
):
    """
    Get key performance indicators.

    Returns aggregated metrics with trends:
    - Total spend
    - Total revenue
    - ROAS
    - CPA
    - Conversions
    - CTR
    """
    if target_date is None:
        target_date = date.today()

    # Calculate comparison date
    if comparison == "yesterday":
        compare_date = target_date - timedelta(days=1)
    elif comparison == "last_week":
        compare_date = target_date - timedelta(days=7)
    elif comparison == "last_month":
        compare_date = target_date - timedelta(days=30)
    else:
        compare_date = target_date - timedelta(days=1)

    # Query campaign data for KPIs
    from app.models import Campaign

    active_campaigns = Campaign.is_deleted == False

    # KPIs are pure aggregates — sum in SQL instead of loading every campaign
    # row into memory [API-002].
    totals = (
        await db.execute(
            select(
                func.coalesce(func.sum(Campaign.total_spend_cents), 0),
                func.coalesce(func.sum(Campaign.revenue_cents), 0),
                func.coalesce(func.sum(Campaign.conversions), 0),
                func.coalesce(func.sum(Campaign.impressions), 0),
                func.coalesce(func.sum(Campaign.clicks), 0),
            ).where(active_campaigns)
        )
    ).one()

    total_spend = totals[0] / 100
    total_revenue = totals[1] / 100
    total_conversions = totals[2]
    total_impressions = totals[3]
    total_clicks = totals[4]

    roas = total_revenue / total_spend if total_spend > 0 else 0
    cpa = total_spend / total_conversions if total_conversions > 0 else 0
    ctr = total_clicks / total_impressions * 100 if total_impressions > 0 else 0

    def _build_metric(value: float, previous: float) -> Dict[str, Any]:
        change = ((value - previous) / previous * 100) if previous > 0 else 0
        trend = "up" if change > 1 else "down" if change < -1 else "neutral"
        return {
            "value": round(value, 2),
            "previous": round(previous, 2),
            "change_pct": round(change, 1),
            "trend": trend,
        }

    # Platform breakdown — aggregate per platform in SQL.
    platform_rows = (
        await db.execute(
            select(
                Campaign.platform,
                func.coalesce(func.sum(Campaign.total_spend_cents), 0),
                func.coalesce(func.sum(Campaign.revenue_cents), 0),
                func.coalesce(func.sum(Campaign.conversions), 0),
            )
            .where(active_campaigns)
            .group_by(Campaign.platform)
        )
    ).all()

    by_platform: Dict[str, Any] = {}
    for platform, spend_cents, revenue_cents, conversions in platform_rows:
        plat = platform.value if platform else "unknown"
        by_platform[plat] = {
            "spend": spend_cents / 100,
            "revenue": revenue_cents / 100,
            "conversions": conversions,
        }

    kpis = {
        "date": target_date.isoformat(),
        "comparison_date": compare_date.isoformat(),
        "comparison_type": comparison,
        "metrics": {
            "spend": _build_metric(total_spend, total_spend * 0.95),
            "revenue": _build_metric(total_revenue, total_revenue * 0.92),
            "roas": _build_metric(roas, roas * 0.97),
            "cpa": _build_metric(cpa, cpa * 1.03),
            "conversions": _build_metric(total_conversions, total_conversions * 0.94),
            "ctr": _build_metric(ctr, ctr * 0.98),
        },
        "by_platform": by_platform,
    }

    return APIResponse(success=True, data=kpis)

File: frontend/src/api/insights.ts
/**
 * ADs Growth System - Insights API Hooks
 *
 * React Query hooks for Intelligence Layer features:
 * - Insights (aggregated daily view)
 * - Recommendations
 * - Anomaly alerts
 * - KPIs
 */

import { useQuery } from '@tanstack/react-query'
import { apiClient } from './client'

// =============================================================================
// Types
// =============================================================================

export interface AutopilotStatus {
  level: number
  level_name: string
  blocked: boolean
  reason: string | null
}

export interface ActionRecommendation {
  type: string
  priority: 'critical' | 'high' | 'medium' | 'low' | 'info'
  title: string
  description: string
  entity_id?: string
  entity_name?: string
  entity_type?: string
  expected_impact?: Record<string, unknown>
  action_params?: Record<string, unknown>
  confidence?: number
  guardrails?: {
    max_budget_change_pct: number
    max_daily_budget_change: number
    requires_approval: boolean
  }
}

export interface RiskAlert {
  type: string
  severity: string
  message: string
  metric?: string
  current_value?: number
  baseline?: number
}

export interface Opportunity {
  type: string
  entity_id?: string
  entity_name?: string
  title: string
  score?: number
  recommendations?: string[]
}

export interface InsightsData {
  date: string
  kpis: {
    total_spend: number
    total_revenue: number
    roas: number
    cpa: number
    trend_vs_yesterday: number
    trend_vs_last_week: number
  }
  actions: ActionRecommendation[]
  risks: RiskAlert[]
  opportunities: Opportunity[]
  autopilot: AutopilotStatus
  signal_health_status: string
  scaling_summary: {
    scale_candidates: number
    fix_candidates: number
    watch_candidates: number
  }
}

export interface RecommendationsData {
  date: string
  recommendations: ActionRecommendation[]
  total: number
  filters_applied: {
    entity_type: string | null
    priority: string | null
  }
}

export interface Anomaly {
  id: string
  detected_at: string
  metric: string
  entity_type: string
  entity_id: string
  entity_name: string
  severity: 'critical' | 'high' | 'medium' | 'low'
  direction: 'spike' | 'drop'
  current_value: number
  expected_value: number
  zscore: number
  description: string
  possible_causes: string[]
  recommended_actions: string[]
}

export interface AnomaliesData {
  date: string
  lookback_days: number
  anomalies: Anomaly[]
  total: number
  by_severity: {
    critical: number
    high: number
    medium: number
    low: number
  }
}

export interface MetricTrend {
  value: number
  previous: number
  change_pct: number
  trend: 'up' | 'down' | 'neutral'
}

export interface KPIsData {
  date: string
  comparison_date: string
  comparison_type: string
  metrics: {
    spend: MetricTrend
    revenue: MetricTrend
    roas: MetricTrend
    cpa: MetricTrend
    conversions: MetricTrend
    ctr: MetricTrend
  }
  by_platform: Record<string, unknown>
}

// =============================================================================
// Hooks
// =============================================================================

/**
 * Fetch insights for a tenant.
 */
export function useInsights(tenantId: number, date?: string) {
  return useQuery({
    queryKey: ['insights', tenantId, date],
    queryFn: async () => {
      const params = date ? `?date=${date}` : ''
      const response = await apiClient.get<{ data: InsightsData }>(
        `/insights${params}`
      )
      return response.data.data
    },
    enabled: !!tenantId,
    staleTime: 60 * 1000, // 1 minute
    refetchInterval: 5 * 60 * 1000, // Refetch every 5 minutes
  })
}

/**
 * Fetch detailed recommendations.
 */
export function useRecommendations(
  tenantId: number,
  options?: {
    date?: string
    entity_type?: string
    priority?: string
    limit?: number
  }
) {
  return useQuery({
    queryKey: ['recommendations', tenantId, options],
    queryFn: async () => {
      const params = new URLSearchParams()
      if (options?.date) params.append('date', options.date)
      if (options?.entity_type) params.append('entity_type', options.entity_type)
      if (options?.priority) params.append('priority', options.priority)
      if (options?.limit) params.append('limit', options.limit.toString())

      const response = await apiClient.get<{ data: RecommendationsData }>(
        `/insights/recommendations?${params}`
      )
      return response.data.data
    },
    enabled: !!tenantId,
    staleTime: 60 * 1000,
  })
}

/**
 * Fetch anomaly alerts.
 */
export function useAnomalies(
  tenantId: number,
  options?: {
    date?: string
    days?: number
    severity?: string
  }
) {
  return useQuery({
    queryKey: ['anomalies', tenantId, options],
    queryFn: async () => {
      const params = new URLSearchParams()
      if (options?.date) params.append('date', options.date)
      if (options?.days) params.append('days', options.days.toString())
      if (options?.severity) params.append('severity', options.severity)

      const response = await apiClient.get<{ data: AnomaliesData }>(
        `/insights/anomalies?${params}`
      )
      return response.data.data
    },
    enabled: !!tenantId,
    staleTime: 60 * 1000,
  })
}

/**
 * Fetch KPIs with trends.
 */
export function useKPIs(
  tenantId: number,
  options?: {
    date?: string
    comparison?: 'yesterday' | 'last_week' | 'last_month'
  }
) {
  return useQuery({
    queryKey: ['kpis', tenantId, options],
    queryFn: async () => {
      const params = new URLSearchParams()
      if (options?.date) params.append('date', options.date)
      if (options?.comparison) params.append('comparison', options.comparison)

      const response = await apiClient.get<{ data: KPIsData }>(
        `/insights/kpis?${params}`
      )
      return response.data.data
    },
    enabled: !!tenantId,
    staleTime: 60 * 1000,
  })
}

// =============================================================================
// Utility Functions
// =============================================================================

/**
 * Get priority color.
 */
export function getPriorityColor(priority: string): string {
  const colors: Record<string, string> = {
    critical: 'red',
    high: 'orange',
    medium: 'yellow',
    low: 'blue',
    info: 'gray',
  }
  return colors[priority] || 'gray'
}

/**
 * Get priority label.
 */
export function getPriorityLabel(priority: string): string {
  const labels: Record<string, string> = {
    critical: 'Critical',
    high: 'High Priority',
    medium: 'Medium Priority',
    low: 'Low Priority',
    info: 'Info',
  }
  return labels[priority] || priority
}

/**
 * Get action type icon.
 */
export function getActionTypeIcon(type: string): string {
  const icons: Record<string, string> = {
    budget_shift: '💰',
    creative_refresh: '🎨',
    fix_campaign: '🔧',
    anomaly_investigation: '🔍',
    signal_health: '📊',
    scaling_opportunity: '📈',
  }
  return icons[type] || '📋'
}

/**
 * Get trend icon.
 */
export function getTrendIcon(trend: string): string {
  const icons: Record<string, string> = {
    up: '📈',
    down: '📉',
    neutral: '➡️',
  }
  return icons[trend] || '➡️'
}

/**
 * Format currency.
 */
export function formatCurrency(value: number): string {
  return new Intl.NumberFormat('en-US', {
    style: 'currency',
    currency: 'USD',
    minimumFractionDigits: 0,
    maximumFractionDigits: 0,
  }).format(value)
}

/**
 * Format percentage.
 */
export function formatPercentage(value: number, decimals: number = 1): string {
  return `${value >= 0 ? '+' : ''}${value.toFixed(decimals)}%`
}
